Packs/ApiModules/Scripts/TAXII2ApiModule/TAXII2ApiModule.py

```diff
@@ -0,0 +1,420 @@
+            trimmed_pattern = pattern.replace(" ", "")
+
+            indicator_groups = self.extract_indicator_groups_from_pattern(
+                trimmed_pattern, self.indicator_regexes
+            )
+            indicators.extend(
+                self.get_indicators_from_indicator_groups(
+                    indicator_groups,
+                    indicator_obj,
+                    STIX_2_TYPES_TO_CORTEX_TYPES,
+                    field_map,
+                )
+            )
+
+            cidr_groups = self.extract_indicator_groups_from_pattern(
+                trimmed_pattern, self.cidr_regexes
+            )
+            indicators.extend(
+                self.get_indicators_from_indicator_groups(
+                    cidr_groups,
+                    indicator_obj,
+                    STIX_2_TYPES_TO_CORTEX_CIDR_TYPES,
+                    field_map,
+                )
+            )
+
+        return indicators
+
+    def get_indicators_from_indicator_groups(
+        self,
+        indicator_groups: List[Tuple[str, str]],
+        indicator_obj: Dict[str, str],
+        indicator_types: Dict[str, str],
+        field_map: Dict[str, str],
+    ) -> List[Dict[str, str]]:
+        """
+        Get indicators from indicator regex groups
+        :param indicator_groups: caught regex group in pattern of: [`type`, `indicator`]
+        :param indicator_obj: taxii indicator object
+        :param indicator_types: supported indicator types -> cortex types
+        :param field_map: map used to create fields entry
+        :return: Indicators list
+        """
+        indicators = []
+        if indicator_groups:
+            for term in indicator_groups:
+                for taxii_type in indicator_types.keys():
+                    # term should be list with 2 argument parsed with regex - [`type`, `indicator`]
+                    if len(term) == 2 and taxii_type in term[0]:
+                        type_ = indicator_types[taxii_type]
+                        value = term[1]
+                        indicator = self.create_indicator(
+                            indicator_obj, type_, value, field_map
+                        )
+                        indicators.append(indicator)
+                        break
+        return indicators
+
+    def extract_indicator_groups_from_pattern(
+        self, pattern: str, regexes: List
+    ) -> List[Tuple[str, str]]:
+        """
+        Extracts indicator [`type`, `indicator`] groups from pattern
+        :param pattern: stix pattern
+        :param regexes: regexes to run to pattern
+        :return: extracted indicators list from pattern
+        """
+        groups: List[Tuple[str, str]] = []
+        for regex in regexes:
+            find_result = regex.findall(pattern)
+            if find_result:
+                groups.extend(find_result)
+        return groups
+
+    @staticmethod
+    def create_indicator(indicator_obj, type_, value, field_map):
```

Comment:
I'm seeing the already built taxii2 feeds we have, MITRE and unit42, and it seems that indicators field mapping is going to be used vastly.
I think we should add a mechanism in this API module that will allow passing these indicatorfields cliname that will be mapped accordingly. WDYT?

Reply:
I agree that's why I pass here a `field_map` to be used for this functionality. Do note the default value that's being passed here comes from `self.field_map` which is being initialized in the `__init__()`, so the way it should work is via a dictionary of {{field_name: field_value}} - the use of this will be demonstrated in a UT